File: 1329-sort-the-matrix-diagonally/1329-sort-the-matrix-diagonally.java
class Solution {
    public int[][] diagonalSort(int[][] mat) {
        int l = mat.length - 1;
        int r = 0;
        int tl = 0;
        int tr = mat.length - 1;
        while (l!=0 || tl!=mat[0].length-1 || r!=mat[0].length-1){
            ArrayList<Integer> list = new ArrayList<>();
            int t1=l;
            int t2=tl;
            while (t1<=tr+1 && t2<=r){
                list.add(mat[t1][t2]);
                t1++;
                t2++;
            }
            int t=0;
            t1=l;
            t2=tl;
            Collections.sort(list);
            while (t1<=tr+1 && t2<=r){
                mat[t1][t2]=list.get(t);
                t++;
                t1++;
                t2++;
            }
            if (l!=0) l--;
            else tl++;
            if (r!=mat[0].length-1) r++;
            else tr--;
        }
        return mat;
    }
}

File: 1329-sort-the-matrix-diagonally/1329-sort-the-matrix-diagonally.py
class Solution:
    def diagonalSort(self, mat: List[List[int]]) -> List[List[int]]:
        l=len(mat)-1
        r=0
        tl=0
        tr=len(mat)-1
        f=True
        while l!=0 or tl!=len(mat[0])-1 or tr!=0 or r!=len(mat[0])-1:
            diag = []
            t1=l
            t2=tl
            while t1<=tr+1 and t2<=r:
                diag.append(mat[t1][t2])
                t1+=1
                t2+=1
            t=0
            t1=l
            t2=tl
            diag.sort()
            while t1<=tr and t2<=r:
                mat[t1][t2]=diag[t]
                t+=1
                t1+=1
                t2+=1
            if l!=0:
                l-=1
            else:
                tl+=1
            if r!=len(mat[0])-1:
                r+=1
            else:
                tr-=1
        return mat